src/utils/Makefile.am

```diff
@@ -1,6 +1,6 @@
 bin_PROGRAMS = cplay crecord
 
-cplay_SOURCES = cplay.c
+cplay_SOURCES = cplay.c wave.c
```

Comment:
Hi the whole series lgtm, but I have some apprehensions on this patch, do we want to support WAV with cplay ? I am not too sure if we would want to go down that path..

@perexg @takaswiedo you guys agree to this ?

Reply:
@vinodkoul thanks for  your reply! I think supporting WAV with cplay is very good for testing compress API implementation both on the Linux kernel side and also on firmware side as its the case with Sound Open Firmware.

If there are any concerns we can try to address them.

Reply:
we have avoided supporting WAV as that conflicts with ALSA pcm.. For PCM support ALSA is the recommended choice.
So am a bit hesitant to merge this. At least we need and @tiwai and @perexg to bless this proposal

Reply:
The code split looks nice. Perhaps, the PCM format may be enabled only conditionally, if someone will compile the code for the debugging purposes from sources. @dbaluta : Could you add configure option for this, please?

Reply:
@perexg sure, I think it works for me to conditionally compile PCM support in ``cplay``. We should keep in mind that ``wave.c`` file will always be compiled in because ``crecord`` always needs it.

Reply:
i think that is fine...